File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/Charm.java
package kz.greetgo.sandbox.controller.model;

public class Charm {
  public Integer id;
  public String name;
  public String description;
  public float energy;

  public Charm() {}

  public Charm(Integer id, String name, String description, float energy) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.energy = energy;
  }
}

File: sandbox.client/front/ts/model/Charm.ts
export class Charm {
  public id: number;
  public name: string;
  public description: string;
  public energy: number;

  public assign(o: any): Charm {
    this.id = o.id;
    this.name = o.name;
    this.description = o.description;
    this.energy = o.energy;
    return this;
  }

  public static copy(a: any): Charm {
    let ret = new Charm();
    ret.assign(a);
    return ret;
  }
}